This notebook contains data exploration about customer groups of Foodish service.

In [2]:
data_path = "C://Users/melania/Documents/Junction/Junction-Kesko-Receipt-Data/Junction_data.csv"

In [3]:
import pandas as pd 
df = pd.read_csv(data_path, nrows=49410457, sep=";") #nrows is explicit due to malforatting of the file

### General data 

In [5]:
N_RECORDS =  len(df)
print("Number of records: ", N_RECORDS)

Number of records:  49410457


In [22]:
data = df

In [9]:
df.head()

AreaId          Receipt TransactionDate  BeginHour            EAN  \
0       1  356601823178935      2017-11-07         17  5410103915654   
1       1  356341113181337      2017-11-05         12  6413466126704   
2       1  356629240622521      2017-11-07         18    24000017677   
3       1  356269217607293      2017-11-04         13  6412000033188   
4       1  356215448049286      2017-11-04         11  2000940900000   

   Quantity PersonAgeGrp  KCustomer QualClass EasyClass  
0  1,000000        55-64       6715     Q_1-3     E_4-7  
1  1,000000        35-44       6712     Q_1-3     E_4-7  
2  1,000000        45-54       6715     Q_1-3     E_4-7  
3  2,000000        55-64       6712     Q_4-7     E_4-7  
4  1,000000        45-54       6713     Q_1-3    E_8-10

In [6]:
receipt_data = df.groupby("Receipt").Quantity.sum()

In [7]:
N_RECEIPTS = len(receipt_data)
print("Number of receipts: ", N_RECEIPTS)

Number of receipts:  5729656


In [14]:
#TIME_DATA 
FIRST_DAY = min(df["TransactionDate"])
LAST_DAY = max(df["TransactionDate"])
print("First date: ", FIRST_DAY, "Last date: ", LAST_DAY)

First date:  2017-11-01 Last date:  2019-09-24


We have 49,410,457 purchases from 5,729,656 receipts, taken between 2017-11-01 and 2019-09-24.

In [20]:
def normalize_data(data): 
    """Retrun number of purchases as fraction of all. 
    Arg: 
    data: data dictionary
    retruns: normalized dictionary. 
    """
    data_norm  = {k: v / N_RECORDS for k, v in data.items()}
    
    return data_norm

In [24]:
#Area data 
area_data = df.AreaId.value_counts()
normalize_data(area_data)

{2: 0.3661986368594,
 1: 0.23175183342262953,
 3: 0.20489767985752488,
 4: 0.19715184986044554}

#### Age data 

In [26]:
age = df.PersonAgeGrp.value_counts()
an = normalize_data(age)
an

{'65-': 0.21617195323653857,
 '45-54': 0.19986961464452757,
 '55-64': 0.1930803432965617,
 '35-44': 0.1893210580910029,
 '25-34': 0.1345485227954884,
 '18-24': 0.026954193117460945}

In [27]:
an['25-34'] + an['18-24']

0.16150271591294935

In early stages of our product, we will concentrate of young poople (18-34), who are most likely to use apps 
and to leave alone and experience problems with bougth but unneeeded food. This groups forms only 16% of purchases, but they are the trend setters and can influence all ages behavior.

### Quality and easiness analises 

In [32]:
q = df.QualClass.value_counts()
qn = normalize_data(q)
qn


{'Q_4-7': 0.4031077267712784,
 'Q_1-3': 0.31362450260275876,
 'Q_8-10': 0.28326777062596287}

In [33]:
print(qn["Q_4-7"]+qn["Q_8-10"])

0.6863754973972412


#### Quality
For customer quality,data  we care about having groups **Q_4_7** and **Q_8_10** as customers.   
The upper group (8-10) consist of the people who realy care about healthy lifestile and enviromental impact. They are likely to already use similar products and are our earlyadopters. 
This groups forms about 28% of the purchases.      
         
The middle group (4-7) contains of people who are seeking for convenience in their lifes. Foodish can help these people save money and 
live in a smarter way. They form 40% of the purchases.    
      
Last group (1-3) is not primary our target, because it includes people who may try to misuse Foodish and take more food than give. 


In [35]:
e = df.EasyClass.value_counts()
en = normalize_data(q)
en


{'Q_4-7': 0.4031077267712784,
 'Q_1-3': 0.31362450260275876,
 'Q_8-10': 0.28326777062596287}

####  Easiness

Fast and easy (8-10) consumers don’t focus on cooking. They are not familiar with cooking and can take advantage of an ap that is showing them recepies. 
They form 31% of purchases. 

Middle group (4-7) are familiar qith basics of cooking, but might be excided to try new things in the neighbourhood. 
They form 31% of purchases. 

Cooking and bakery (1-3) customers prefer are masters in the kitchen. They might be interested to meet new people through food and share there art. 
They form 28% of purchases. 


### KCustomer types

In [39]:
k = df.KCustomer.value_counts()
kn = normalize_data(k)
kn

{6714: 0.24068896994820346,
 6713: 0.20415478043443314,
 6711: 0.197529199132888,
 6712: 0.18546669179764924,
 6715: 0.17216035868682616}

In [42]:
kn[6714] + kn[6713] + kn[6713]

0.6489985308170697

KCustomer type values correspond to:   
6711 - Enthusiast                 
6712 - Pleasure-seeking       
6713 - Aware          
6714 - Convenience-seeking  
6715 - Established       
   
Our targets are enthusiast, who might end with two much food after shopping, people who seek convenience, and people who are aware about their shoppig behavior. 
They form totally 65% of purchases. 